common/src/main/java/com/aliyun/emr/rss/common/network/util/MemoryTracker.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.common.network.util;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import io.netty.util.internal.PlatformDependent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.VM;
+
+public class MemoryTracker {
+  private long maxDirectorMemory = VM.maxDirectMemory();
+  private long high = 0;
+  private long low = 0;
+  private long directMemoryCritical = 0;
+  private AtomicLong storageMemoryTracker = new AtomicLong();
```

Comment:
I think storageMemoryTracker can be deleted and directMemoryIndicator is enough to indicate whether direct memory's pressure. The memory of storageMemoryTracker is just part of directMemoryIndicator since the increment() method just adds up the netty's memory.